### PR TITLE
expand name search & fix login-mechanism reporting

### PR DESCRIPTION
**Pull request by JoeArmani**
This PR is in support of:
https://github.com/episphere/connect/issues/654
And
https://github.com/episphere/dashboard/pull/538

Re: Expand name search:
•Write updates to `query.firstName` and `query.lastName` fields as arrays.
`query.firstName` field includes: consent first name (471168198), first name (399159511), and preferred name (153211406).
`query.lastName` field includes: consent last name (736251808) and last name(996038075).

Re: Multi-logins:
•Fix handling of `phone`, `password`, and `passwordAndPhone` sign-in mechanism reporting (995036844). The issue was: ‘noreply’ emails were not factored into the reporting functionality, so removed emails were causing incorrect reporting of sign-in mechanism.

•Hide ‘noreply’ account emails in the participant lookup results. I am now aware that this field exists in results and the results are hidden when accountEmail starts with ‘noreply’. Screenshots (before and after) are included.

Before:
before noreply filter

After:
after noreply filter

### CODE REVIEW

File: siteManagerDashboard/participantCommons.js [hunk 1]
```diff
@@ -812,6 +814,7 @@ const tableTemplate = (data, showButtons) => {
 }
 
 const addEventShowMoreInfo = (data) => {
+    console.log('addEventShowMoreInfo');
```
Comment (abhinavjonnada82) [inline siteManagerDashboard/participantCommons.js:817]: remove `console.log`

File: siteManagerDashboard/participantDetailsHelpers.js [hunk 2]
```diff
@@ -1131,6 +1131,8 @@ const cleanPhoneNumber = (changedOption) => {
 
 };
 
+const firstNameTypes = [fieldMapping.fName, fieldMapping.prefName, fieldMapping.consentFirstName];
```
Comment (abhinavjonnada82) [inline siteManagerDashboard/participantDetailsHelpers.js:1134]: 👍

File: siteManagerDashboard/participantLookup.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import { renderNavBarLinks, dashboardNavBarLinks, renderLogin, removeActiveClass } from './navigationBar.js';
```
Comment (abhinavjonnada82) [inline siteManagerDashboard/participantLookup.js:1]: Hoping this are unused functions. In that case thanks

File: siteManagerDashboard/participantDetailsHelpers.js [hunk 4]
```diff
@@ -1166,6 +1170,34 @@ export const submitClickHandler = async (participant, changedOption) => {
     }
 };
 
+/**
+ * Handle the query.frstName and query.lastName fields in the participant profile.
```
Comment (abhinavjonnada82) [inline siteManagerDashboard/participantDetailsHelpers.js:1174]: I agree these descriptive notes are helpful but I suggest you condense them 1-2 lines or just remove it entirely.

**Review by abhinavjonnada82 — APPROVED** [review]

**Comment by abhinavjonnada82** [comment]
Minor suggestions

File: siteManagerDashboard/participantLookup.js (hunk 3 above)
Comment (JoeArmani) [reply siteManagerDashboard/participantLookup.js:1]: They are!

File: siteManagerDashboard/participantCommons.js (hunk 1 above)
Comment (JoeArmani) [reply siteManagerDashboard/participantCommons.js]: done.

**Review by we-ai — APPROVED** [review]
Looks good.

**Comment by JoeArmani** [comment]
removed console.log() for the query string. Don't want to see this outside of local env.

**Pull request merged**